### ISSUE TITLE
Error: awx.main.dispatch Worker failed to run task awx.main.tasks.jobs.RunJob

### ISSUE THREAD

**Issue by nimeshjain1** [ansible/awx#13647]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Bug Summary

We have been running ansible jobs and there are intermittent issues where the job fails without any output. On looking into the error in the awx dashboard we see the issue (Task was maked as running but was not present in the job queue, so it has been marked as failed. ) On debugging this further we realise that the kubernetes pod is not created but we do find the following error in the awx logs:

awx.main.dispatch Worker failed to run task awx.main.tasks.jobs.RunJob

All of these jobs are running as scheduled jobs and they fail restart resulting into poor confidence on the success of these jobs.

### AWX version

21.6.0

### Select the relevant components

- [ ] UI
- [X] API
- [ ] Docs
- [X] Collection
- [ ] CLI
- [ ] Other

### Installation method

kubernetes

### Modifications

no

### Operating system

Amazon Linux 2

### Web browser

Chrome

### Steps to reproduce

This is intermittent issue, It happens on any of the job. We have been running multiple jobs multiple times. Next run of the same job succeeds wihout any error

### Expected results

All jobs should succed.

### Actual results

Failed with the error((Task was maked as running but was not present in the job queue, so it has been marked as failed.)  without any output.

**Comment by hostmit** [comment]
It worth mentioning, that we are using AWX operator, jobs are running on container groups in EKS.
Upon checking logs, by job ID, the ones that error with message above have no logs in K8S api server. This means EKS never got a request to create a corresponding job, must be failing somewhere between AWX app and EKS.

**Comment by mani3887** [comment]
I am able to reproduce this condition easily by sending more load to the AWX system. Are you observing this condition when the system is under load as well?

**Comment by nimeshjain1** [comment]
That is right, we suspect this to be a case when the load is high.

**Comment by djyasin** [comment]
@nimeshjain1 thank you for submitting this information.
We would like to gather a bit more information from you so that we can better identify what is occurring here. 

How many jobs are you launching against your container group, and what does your cluster look like?

**Comment by fosterseth** [comment]
might be worth setting the `max concurrent jobs` on the container_group where the jobs are running

![image](https://user-images.githubusercontent.com/8745776/223806908-0dc542d8-b3ae-43d3-bbcd-9bbe73b233ff.png)

**Comment by mani3887** [comment]
@fosterseth : Thanks for your response. In my version of AWX, I do not see max concurrent jobs; but when I looked into the code, I observed start_task_limit which I set to 70; and set the awx replicas to 5, with each having 7 CPUs and 24Gi; I was assuming I would be able to run at least 200 jobs; but that failed with the same error.

**Comment by nimeshjain1** [comment]
Not sure which version is this released even i am unable to see the option of setting the max current jobs on the container_groups. We are uisng the 12.6.0 version of AWX.  As such we have been running quite a few scheduled jobs and our suspicion is we hit this issue when we are running around 90 jobs or so at any point in time. The failing of  the jobs are very random and we are unable to corelate with the db performance also to check if the issue is happening because of the performance on the db side.

**Comment by fosterseth** [comment]
yeah this param landed in 21.10.0 @mani3887  https://github.com/ansible/awx/releases/tag/21.10.0

**Comment by hostmit** [comment]
Hello!

I'd like to update this issue so it may be of use to someone else.

In our case (myself and @nimeshjain1 work at the same company), this error has also a dns resolution message side by side.

` "log": "psycopg2.OperationalError: could not translate host name \"db1.ansible.<redacted>\" to address: Name or service not known\n",
`
VPC resolver query logging showed no error. "core-dns" component was hosted in fargate. It seems to be a problem. As soon we moved it to managed nodes, error disappeared.

I think it is safe to close this issue.

**Comment by mani3887** [comment]
Thanks for the update. Where do you see this error message? Do you see this error message in awx task? @hostmit , @nimeshjain1
